net/http/isvalidcookievalue.c

```diff
@@ -0,0 +1,39 @@
+/*-*- mode:c;indent-tabs-mode:nil;c-basic-offset:2;tab-width:8;coding:utf-8 -*-│
+│vi: set net ft=c ts=2 sts=2 sw=2 fenc=utf-8                                :vi│
+╞══════════════════════════════════════════════════════════════════════════════╡
+│ Copyright 2021 Justine Alexandra Roberts Tunney                              │
+│                                                                              │
+│ Permission to use, copy, modify, and/or distribute this software for         │
+│ any purpose with or without fee is hereby granted, provided that the         │
+│ above copyright notice and this permission notice appear in all copies.      │
+│                                                                              │
+│ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL                │
+│ WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED                │
+│ WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE             │
+│ AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL         │
+│ DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR        │
+│ PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER               │
+│ TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR             │
+│ PERFORMANCE OF THIS SOFTWARE.                                                │
+╚─────────────────────────────────────────────────────────────────────────────*/
+#include "libc/str/str.h"
+#include "net/http/http.h"
+
+/**
+ * Returns true if string is a valid cookie value
+ * (any ASCII excluding control char, whitespace,
+ * double quotes, comma, semicolon, and backslash).
+ *
+ * @param n if -1 implies strlen
+ */
+bool IsValidCookieValue(const char *s, size_t n) {
+  size_t i;
+  if (!n) return false;
+  if (n == -1) n = s ? strlen(s) : 0;
+  for (i = 0; i < n; ++i) {
+    if (!kHttpCookieValue[s[i] & 0xff]) {
```

Comment:
I'd recommend the following code since the generated assembly looks faster and tinier than a lookup table:

```c
static inline bool IsCookieOctet(unsigned char i) {
  return i > 0x20 && i < 0x7F && i != ',' && i != ';' && i != '\\' &&
         i != ' ' && i != '"';
}
```

Reply:
Good point; I'll add this to redbean.c directly and remove the added table/method files. Thanks!

Reply:
Unless you think it's still worthwhile to have `isvalidcookievalue.c `

Reply:
redbean.c is a monster in terms of lines of code. If it can be easily made a separate file then I'd recommend doing so.

Reply:
Yes, I kept this logic separate.